test/tests/S6ImportExportTests.cpp

```diff
@@ -0,0 +1,570 @@
+#include <openrct2/network/network.h>
+#include <openrct2/object/ObjectManager.h>
+#include <openrct2/platform/platform.h>
+#include <openrct2/rct2/S6Exporter.h>
+#include <openrct2/ride/Ride.h>
+#include <openrct2/world/Park.h>
+#include <openrct2/world/Sprite.h>
+#include <stdio.h>
+#include <string>
+
+using namespace OpenRCT2;
+
+struct GameState_t
+{
+    rct_sprite sprites[MAX_SPRITES];
+};
+
+static bool LoadFileToBuffer(MemoryStream& stream, const std::string& filePath)
+{
+    FILE* fp = nullptr;
+    fopen_s(&fp, filePath.c_str(), "rb");
+    EXPECT_NE(fp, nullptr);
+    if (fp == nullptr)
+        return false;
+
+    uint8_t buf[1024];
+    size_t bytesRead = fread(buf, 1, sizeof(buf), fp);
+    while (bytesRead > 0)
+    {
+        stream.Write(buf, bytesRead);
+        bytesRead = fread(buf, 1, sizeof(buf), fp);
+    }
+    fclose(fp);
+
+    return true;
+}
+
+static void GameInit(bool retainSpatialIndices)
+{
+    if (!retainSpatialIndices)
+        reset_sprite_spatial_index();
+
+    reset_all_sprite_quadrant_placements();
+    scenery_set_default_placement_configuration();
+    load_palette();
+    map_reorganise_elements();
+    sprite_position_tween_reset();
+    AutoCreateMapAnimations();
+    fix_invalid_vehicle_sprite_sizes();
+
+    gGameSpeed = 1;
+}
+
+static bool ImportSave(MemoryStream& stream, std::unique_ptr<IContext>& context, bool retainSpatialIndices)
+{
+    stream.SetPosition(0);
+
+    auto& objManager = context->GetObjectManager();
+
+    auto importer = ParkImporter::CreateS6(context->GetObjectRepository());
+    auto loadResult = importer->LoadFromStream(&stream, false);
+    objManager.LoadObjects(loadResult.RequiredObjects.data(), loadResult.RequiredObjects.size());
+    importer->Import();
+
+    GameInit(retainSpatialIndices);
+
+    return true;
+}
+
+static bool ExportSave(MemoryStream& stream, std::unique_ptr<IContext>& context)
+{
+    auto& objManager = context->GetObjectManager();
+
+    auto exporter = std::make_unique<S6Exporter>();
+    exporter->ExportObjectsList = objManager.GetPackableObjects();
+    exporter->Export();
+    exporter->SaveGame(&stream);
+
+    return true;
+}
+
+static std::unique_ptr<GameState_t> GetGameState(std::unique_ptr<IContext>& context)
+{
+    std::unique_ptr<GameState_t> res = std::make_unique<GameState_t>();
+    for (size_t spriteIdx = 0; spriteIdx < MAX_SPRITES; spriteIdx++)
+    {
+        rct_sprite* sprite = get_sprite(spriteIdx);
+        if (sprite == nullptr)
+            res->sprites[spriteIdx].generic.sprite_identifier = SPRITE_IDENTIFIER_NULL;
+        else
+            res->sprites[spriteIdx] = *sprite;
+    }
+    return res;
+}
+
+static void AdvanceGameTicks(uint32_t ticks, std::unique_ptr<IContext>& context)
+{
+    auto* gameState = context->GetGameState();
+    for (uint32_t i = 0; i < ticks; i++)
+    {
+        gameState->UpdateLogic();
+    }
+}
+
+#define COMPARE_FIELD(field) ASSERT_EQ(left.field, right.field)
```

Comment:
Why assert? The test could continue to find all mismatched fields with expect

Reply:
Can change it to EXPECT_EQ